lib/Sabre/Album/AlbumPhoto.php

```diff
@@ -0,0 +1,99 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * @copyright Copyright (c) 2021 Robin Appelman <robin@icewind.nl>
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Photos\Sabre\Album;
+
+use OCA\Photos\Album\AlbumFile;
+use OCA\Photos\Album\AlbumInfo;
+use OCA\Photos\Album\AlbumMapper;
+use OCP\Files\Folder;
+use OCP\Files\Node;
+use OCP\Files\File;
+use OCP\Files\NotFoundException;
+use Sabre\DAV\Exception\Forbidden;
+use Sabre\DAV\IFile;
+
+class AlbumPhoto implements IFile {
+	private AlbumMapper $albumMapper;
+	private AlbumInfo $album;
+	private AlbumFile $file;
+	private Folder $userFolder;
+
+	public function __construct(AlbumMapper $albumMapper, AlbumInfo $album, AlbumFile $file, Folder $userFolder) {
+		$this->albumMapper = $albumMapper;
+		$this->album = $album;
+		$this->file = $file;
+		$this->userFolder = $userFolder;
+	}
+
+	public function delete() {
+		$this->albumMapper->removeFile($this->album->getId(), $this->file->getFileId());
+	}
+
+	public function getName() {
+		return $this->file->getFileId() . "-" . $this->file->getName();
```

Comment:
Why include the file id ?

Reply:
Oh, to not have two files with the same name in an album ?

Reply:
Yes